### DISCUSSION TITLE
Obtaining angles from rotation matrix

### DISCUSSION THREAD

**Discussion by Abhishekc95** [neperfepx/neper#502] [Q&A] [answered]
I have obtained rotation matrix from neper tessellation files, but i need to convert to angles in sequence of ZXY (first rotation about local z then local x and then y).
or is there anyway i can convert other ori formats to ZXY sequence.
Im trying input these angles to element orientation in Ansys apdl software.
i apologize in advance as this question is not related directly to neper.

**Comment by rquey** [comment, marked as answer]
This is not a known convention to me, and not implemented in Neper.  You'll have to find the formulae to go from the rotation matrix to the angles.

**Reply by Abhishekc95** [reply]
Just for clarification, from the discussion page I got to know that each row in the rotation matrix represents vectors parallel to the crystallographic axes. how does neper calculate this matrix.

**Reply by rquey** [reply]
I can only point you to Orilib's documentation, https://sourceforge.net/projects/orilib.

**Reply by Abhishekc95** [reply]
thanks a lot
